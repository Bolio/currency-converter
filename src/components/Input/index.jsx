import React from "react";
import TextField from "@mui/material/TextField";

const Input = ({ id, label, type, focus, change, value }) => (
  <TextField
    id={id}
    label={label}
    variant="filled"
    fullWidth
    type={type}
    onFocus={focus}
    onChange={change}
    value={value}
  />
);

export default Input;
