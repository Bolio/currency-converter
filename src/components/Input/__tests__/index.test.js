import React from "react";
import Input from "..";
import { render } from "@testing-library/react";

describe("Input component", () => {
  const component = (
    <Input
      id="currency-moneda"
      label="MONEY-TEST"
      type="number"
      focus={jest.fn()}
      change={jest.fn()}
      value={371.84}
    />
  );

  test("Should render Input component", () => {
    const { container } = render(component);
    expect(container).toMatchSnapshot();
  });
});
