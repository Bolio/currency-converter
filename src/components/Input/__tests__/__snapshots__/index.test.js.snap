// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input component Should render Input component 1`] = `
<div>
  <div
    class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
  >
    <label
      class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-filled MuiFormLabel-root MuiFormLabel-colorPrimary MuiFormLabel-filled css-o943dk-MuiFormLabel-root-MuiInputLabel-root"
      data-shrink="true"
      for="currency-moneda"
      id="currency-moneda-label"
    >
      MONEY-TEST
    </label>
    <div
      class="MuiFilledInput-root MuiFilledInput-underline MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-gjbq6i-MuiInputBase-root-MuiFilledInput-root"
    >
      <input
        aria-invalid="false"
        class="MuiFilledInput-input MuiInputBase-input css-10botns-MuiInputBase-input-MuiFilledInput-input"
        id="currency-moneda"
        type="number"
        value="371.84"
      />
    </div>
  </div>
</div>
`;
